File: hmbackend/src/Api/Features/Organizations/OrganizationViewModel.cs
namespace Api.Features.Organizations;

public class OrganizationViewModel
{
    public required Guid OrganizationId { get; init; }

    public required IEnumerable<ProjectLookup> Projects { get; init; }

    public class ProjectLookup
    {
        public required Guid Id { get; init; }

        public required string Name { get; init; }

        public required string Owner { get; init; }

        public required decimal ProgressPercentage { get; init; }
    }
}

File: hmclient/src/pages/organizations/OrganizationViewModel.ts
import {z} from 'zod';

export const ProjectModelSchema =
    z.object({
        id: z.string(),
        name: z.string(),
        description: z.string(),
        owner: z.string(),
        progressPercentage: z.number()
    })

export const OrganizationViewModelSchema =
    z.object({
        organizationId: z.string(),
        projects: z.array(
            ProjectModelSchema)

    })

export type OrganizationViewModel = z.infer<typeof OrganizationViewModelSchema>;

export type ProjectModel = z.infer<typeof ProjectModelSchema>;
